### PR TITLE
build: simplify actions

### PR DESCRIPTION
**Pull request by zzmp**
- Simplifies github actions (CI/CD)
- Introduces reusable actions (to be extracted to a separate infra repo)

### CODE REVIEW

File: .github/workflows/check-pr-title.yaml
Comment (vm) [inline .github/workflows/check-pr-title.yaml:11]: not sure what semantic-commit is

File: .github/workflows/release.yaml
Comment (vm) [inline .github/workflows/release.yaml:13]: can you tell me what this flag does?

File: .github/workflows/test.yml
Comment (vm) [inline .github/workflows/test.yml:11]: not sure what this is

**Review by vm — APPROVED** [review]

File: .github/workflows/check-pr-title.yaml
Comment (zzmp) [reply .github/workflows/check-pr-title.yaml:11]: this should be [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/), because that is what it is checking.

File: .github/workflows/release.yaml
Comment (zzmp) [reply .github/workflows/release.yaml:13]: this only waits for the first check suite. this is meant to wait for the test suite generated on merge or push, as opposed to including any suites that are run afterwards, which may include other things (crowdin, or other failed releases).

File: .github/workflows/test.yml
Comment (zzmp) [reply .github/workflows/test.yml:11]: this gives maintainers the ability to run the test check suite manually, so that it can be run against branches besides main on a case-by-case basis.

**Pull request merged**
